### PR TITLE
Support returning multiple polygons

### PR DESCRIPTION
**Pull request by orangejulius**
Contains the changes from #10, rebased on top of master to resolve conflicts, and with tests rewritten for clarity.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,20 +24,25 @@ function PolygonLookup( featureCollection ){
...
-PolygonLookup.prototype.search = function search( x, y ){
+PolygonLookup.prototype.search = function search( x, y, limit ){
+  if ( limit === 0 ){
+    return [];
+  }
   var bboxes = this.rtree.search( { minX: x, minY: y, maxX: x, maxY: y } );
   var pt = [ x, y ];
-  for( var ind = 0; ind < bboxes.length; ind++ ){
+  var results = [];
+  for( var ind = 0; ind < bboxes.length && ( limit === undefined || limit === -1 || limit > 0 ); ind++ ){
```
Comment (trescube) [inline index.js]: This is really hard to read and leads to some confusing state later.  Could limit be overridden with `MAX_SAFE_INTEGER` when it's initial value is `-1`?

Comment (orangejulius) [inline index.js]: Good idea. I've taken a second to make the logic cleaner, take a look. (it's still not perfect)

**Comment by orangejulius** [comment]
Note: I think that while this is fine to merge for now, we should consider a breaking change release where the search function always returns an array of polygon results up to a certain limit. This simplifies the logic a lot. Most places in Pelias probably eventually want to handle multiple point in polygon results anyway.

**Comment by dianashk** [comment]
LGTM

File: index.js [hunk 2]
```diff
@@ -23,36 +24,94 @@ function PolygonLookup( featureCollection ){
...
+      }
+    }
+    return true;
+  }
+  return false;
+}
+
+
+/*
+ * Interrnal helper method to return a single matching polygon
```
Comment (trescube) [inline index.js]: `Internnal` -> `Internal`

**Comment by trescube** [comment]
egregious typo, otherwise :shipit:

**Comment by dianashk** [comment]
LGTM

**Pull request merged**